rules/static_rule_test.go

```diff
@@ -82,69 +82,75 @@ func verifyTestAttributes(t *testing.T, rule staticRule) {
+	factory := newCompareLiteralRuleFactory(newEqualsComparator(&ruleValue), "%s = val1")
 	rule := factory.newRule([]string{"/prefix/mykey"}, getTestAttributes())
 	queryValue := "val1"
 	result := rule.satisfiable("/prefix/mykey", &queryValue)
 	assert.True(t, result)
 	verifyTestAttributes(t, rule)
+	assert.Equal(t, "/prefix/mykey = val1", rule.String())
+	factory = newCompareLiteralRuleFactory(newEqualsComparator(nil), "%s = <nil>")
+	rule = factory.newRule([]string{"/prefix/mykey"}, getTestAttributes())
+	assert.Equal(t, "/prefix/mykey = <nil>", rule.String())
 }
 
-func TestEqualsLiteralError(t *testing.T) {
+func TestCompareLiteralError(t *testing.T) {
 	ruleValue := "val1"
-	factory := newEqualsLiteralRuleFactory(&ruleValue)
+	factory := newCompareLiteralRuleFactory(newEqualsComparator(&ruleValue), "%s = val1")
 	rule := factory.newRule([]string{"/prefix/mykey"}, getTestAttributes())
 	_, err := rule.satisfied(&errorAPI)
 	assert.Equal(t, errAPI, err)
 }
 
-func TestEqualsLiteralEqualsNil(t *testing.T) {
-	rule := equalsLiteralRule{
-		key:   "/prefix/mykey",
-		value: nil,
+func TestCompareLiteralEqualsNil(t *testing.T) {
+	rule := compareLiteralRule{
+		key:            "/prefix/mykey",
+		comparator:     newEqualsComparator(nil),
+		stringTemplate: "%s = <nil>",
 	}
 	result := rule.satisfiable("/prefix/mykey", nil)
 	assert.True(t, result)
+	assert.Equal(t, "/prefix/mykey = <nil>", rule.String())
 }
 
-func TestEqualsLiteralKeyMismatch(t *testing.T) {
+func TestCompareLiteralKeyMismatch(t *testing.T) {
 	ruleValue := "val1"
 	queryValue := "val1"
-	rule := equalsLiteralRule{
-		key:   "/prefix/mykey1",
-		value: &ruleValue,
+	rule := compareLiteralRule{
+		key:        "/prefix/mykey1",
+		comparator: newEqualsComparator(&ruleValue),
 	}
 	result := rule.satisfiable("/prefix/mykey2", &queryValue)
 	assert.False(t, result)
 	assert.False(t, rule.keyMatch("val2"))
 }
 
-func TestEqualsLiteralOnlyRuleNil(t *testing.T) {
+func TestCompareLiteralOnlyRuleNil(t *testing.T) {
 	queryValue := "val1"
-	rule := equalsLiteralRule{
-		key:   "/prefix/mykey",
-		value: nil,
+	rule := compareLiteralRule{
+		key:        "/prefix/mykey",
+		comparator: newEqualsComparator(nil),
 	}
 	result := rule.satisfiable("/prefix/mykey", &queryValue)
 	assert.True(t, result)
 }
 
-func TestEqualsLiteralOnlyQueryNil(t *testing.T) {
+func TestCompareLiteralOnlyQueryNil(t *testing.T) {
 	ruleValue := "val1"
-	rule := equalsLiteralRule{
-		key:   "/prefix/mykey",
-		value: &ruleValue,
+	rule := compareLiteralRule{
+		key:        "/prefix/mykey",
+		comparator: newEqualsComparator(&ruleValue),
 	}
 	result := rule.satisfiable("/prefix/mykey", nil)
 	assert.True(t, result)
 }
 
-func TestEqualsLiteralFactory(t *testing.T) {
+func TestCompareLiteralFactory(t *testing.T) {
 	value := "val1"
-	factory := equalsLiteralRuleFactory{
-		value: &value,
+	factory := compareLiteralRuleFactory{
+		comparator: newEqualsComparator(&value),
```

Comment:
Should there be a new test case using some test Comparator function instead of always using the built-in newEqualsComparator one?

Reply:
I thought about that but came to the conclusion that the existing test cases for the equals literal rule test the comparison functionality as much as is possible: equal, unequal and `nil` values are passed in and both `true` and `false` are returned. I can't think of any other testing I'd do.